webserver/cors/src/main/java/io/helidon/webserver/cors/LogHelper.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package io.helidon.webserver.cors;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.logging.Level;
+
+import io.helidon.common.http.Http;
+import io.helidon.webserver.cors.CORSSupport.RequestAdapter;
+import io.helidon.webserver.cors.CORSSupportHelper.RequestType;
+
+import static io.helidon.common.http.Http.Header.HOST;
+import static io.helidon.common.http.Http.Header.ORIGIN;
+import static io.helidon.webserver.cors.CORSSupportHelper.LOGGER;
+import static io.helidon.webserver.cors.CrossOriginConfig.ACCESS_CONTROL_REQUEST_METHOD;
+
+class LogHelper {
+
+    static final Level DECISION_LEVEL = Level.FINE;
+
+    private LogHelper() {
+    }
+
+    /**
+     * Collects headers for assignment to a request or response and logging during assignment.
+     */
+    static class Headers {
+        private final List<Map.Entry<String, Object>> headers = new ArrayList<>();
+        private final List<String> notes = LOGGER.isLoggable(DECISION_LEVEL) ? new ArrayList<>() : null;
+
+        Headers add(String key, Object value) {
+            headers.add(new AbstractMap.SimpleEntry<>(key, value));
+            return this;
+        }
+
+        Headers add(String key, Object value, String note) {
+            add(key, value);
+            if (notes != null) {
+                notes.add(note);
+            }
+            return this;
+        }
+
+        void set(BiConsumer<String, Object> consumer, String note) {
+            headers.forEach(entry -> consumer.accept(entry.getKey(), entry.getValue()));
+            LOGGER.log(DECISION_LEVEL, () -> note + ": " + headers + (notes == null ? "" : notes));
+        }
+    }
+
+    static <T> boolean isRequestTypeNormal(boolean result, RequestAdapter<T> requestAdapter, Optional<String> originOpt,
```

Comment:
This does not seem to be a method relevant to the name of the class.

Reply:
As it is only used to log information, the return type should be `void` and maybe rename it to something log relevant?